### ISSUE TITLE
Generate `std::ffi::*` instead of the `std::os::raw::*` types

### ISSUE THREAD

**Issue by nyurik** [rust-lang/rust-bindgen#2924]
[`std::os::raw`](https://doc.rust-lang.org/std/os/raw/) recommends to use `core::ffi` instead, but for std crates using core directly seem to trigger some warnings.  The `std::ffi::*` offers the same types like `c_char` and `c_int` as `core::ffi::*`, plus it offers the newer types like `CStr` in there. Would it make sense to consistently use just the `std::ffi::*` for all C types in the generated code?

**Comment by pvdrz** [comment]
I think `core::ffi` is used if the `use_core` option is enabled and the rust target is high enough. Isn't that the case?

**Comment by GKFX** [comment]
It would be helpful to default to `core::ffi` on editions/versions that support it, since that should work on both std and no-std environments and is one less setting to have to set.

**Comment by pvdrz** [comment]
I think this is a sensible default. I don't see any immediate reason to not use `core::ffi` if available.

In general I don't see any reasons to not use `core` instead of `std` in general (unless it is something that `std` offers and `core` does not, of course).

**Comment by MarijnS95** [comment]
Yeah, seems nice to default this to `std::ffi::` if `--rust-target 1.64` or higher (and `core::` if `--use-core` is specified, of course).
